README.md

```diff
@@ -10,3 +10,62 @@ disk is running low on inodes or space.
 The script has initially been developed to help installations of BinderHub clean
 up space on nodes as it otherwise can run out of space and stop being able to
 build now docker images.
+
+## Why?
+
+Container images are one of the biggest consumers of disk space
+and inodes on kubernetes nodes. Kubernetes tries to make sure there is enough
+disk space on each node by [garbage
+collecting](https://kubernetes.io/docs/concepts/architecture/garbage-collection/#containers-images)
+unused container images and containers. Tuning this is important
+for [binderhub](https://github.com/jupyterhub/binderhub/) installations,
+as many images are built and used only a couple times. However, on
+most managed kubernetes installations (like GKE, EKS, etc), we can not
+tune these parameters!
```

Comment:
Ohh so docker-image-cleaner is cleaning up images pulled by `kubelet` to the node, not just the cache etc of images relevant for the builder pods that handle a separate set of docker images?

I think of two things, and don't dare assume they are the same, or what docker-image-cleaner is purposed for.

1. Each node's images pulled by `kubelet` into the CRI used by k8s.
2. Each node's build pod images, pulled by the builder pods relying on a `dind` daemonset that expose some docker socket with associated local image cache.

I guess further this plays into a confusion about `dind` in the BinderHub Helm chart as well. `dind` in BinderHub is something one opts into, right? What does it mean to not use dind in the BinderHub Helm chart? I'm generally a bit confused here about `dind` I assume.

Reply:
Ah, found the text below to probably answer my confusion. There are assumptions about the node's where this run, that this should run once per node in a k8s cluster with a `docker` CRI.

```
1. Only kubernetes nodes using the `docker` runtime are supported.
   `containerd` or `cri-o` container backends are not supported.
```

Reply:
Hmmm, since I can't think of a clear suggestion on what to add etc at this point. I think its reasonable to go for a merge and iterate from there.